src/braket/circuits/free_parameter.py

```diff
@@ -0,0 +1,112 @@
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from numbers import Number
+
+from sympy import Symbol
+
+
+class FreeParameter:
+    """
+    Class 'FreeParameter'
+    """
+
+    def __init__(self, name: str):
+        """
+        Initializes a new :class:'FreeParameter' object.
+        Free parameters can be used in parameterized circuits.
+
+        Args:
+            name: Name of the :class:'FreeParameter'. Can be a unicode value.
+
+        Examples:
+            >>> param1 = FreeParameter("theta")
+            >>> param1 = FreeParameter("\u03B8")
+        """
+        parameter_symbol = Symbol(name)
+        self._name = name
+        self._symbol = parameter_symbol
+        self._parameter_expression = parameter_symbol
+        self._parameter_value = Number
+
+    @property
+    def name(self) -> str:
+        """
+        Returns:
+            str: Name of this parameter.
+        """
+        return self._name
+
+    @property
+    def parameter_value(self) -> Number:
+        """
+        This is set to None until a value is assigned.
+
+        Returns:
+            Number: The value assigned to a parameter.
+        """
+        return self._parameter_value
+
+    @property
+    def symbol(self) -> Symbol:
+        """
+        Returns:
+            Symbol: The symbol for a parameter.
+        """
+        return self._symbol
+
+    def fix_value(self, param_value: Number) -> None:
+        """
+        Sets the value for a :class:'FreeParameter'.
+
+        Args:
+            param_value: The value to be set for the object.
+        """
+        self._validate_values(param_value)
+        self._parameter_value = float(param_value)
+
+    def _validate_values(self, param_value):
+        """
+        Validates that the param_value being passed in is numeric. Raises a ValueError otherwise.
+
+        Args:
+            param_values: A value to be checked.
+
+        Raises:
+            ValueError: If the param_value is not a Number.
+
+        """
+        if not isinstance(param_value, Number):
+            raise ValueError(
+                f"Parameters value assignment can only take numeric values. "
+                f"Invalid inputs: {param_value}"
+            )
+
+    def __str__(self):
+        return self.name
+
+    def __hash__(self) -> int:
+        return hash(tuple(self._name))
+
+    def __eq__(self, other):
+        if isinstance(other, FreeParameter):
+            return self.name == other.name
+        return False
+
+    def __repr__(self):
+        """
+        The representation of the :class:'FreeParameter'.
+
+        Returns: The name of the class:'FreeParameter' to represent the class.
+        """
+        return self.name
```

Comment:
It might be niche, but there are use cases that the angle of a gate is a product of two variables. Would it easy to also add multiplication of `free_parameter`? 

A use case is when users try to implement a variation of [this circuit](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.IQPEmbedding.html). Without multiplication, user need to define `x1*x2` as a free parameter in addition to `x1` and `x2`.

Reply:
Currently, without expressions acting on free_parameter, this would have to be done by using multiplication for the values of `x1` and `x2`. In the future, Something such as `rX(x1x2)` would be supported instead of having to pass values in manually.